dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/MetadataDumper.java

```diff
@@ -183,6 +192,33 @@ private void print(@Nonnull Task<?> task, int indent) {
     }
   }
 
+  private Path prepareOutputPath(
+      @Nonnull String fileName, @Nonnull Closer closer, @Nonnull ConnectorArguments arguments)
+      throws IOException {
+    Matcher matcher = GCS_PATH_PATTERN.matcher(fileName);
+    if (matcher.matches()) {
+      String bucket = matcher.group("bucket");
+      String path = matcher.group("path");
+      LOG.debug(
+          String.format(
+              "Setting up CloudStorageFileSystem with bucket '%s' and path '%s'.", bucket, path));
+      CloudStorageFileSystem cloudStorageFileSystem =
+          closer.register(CloudStorageFileSystem.forBucket(bucket));
```

Comment:
I cannot test it on a cloudtop because I cannot revoke my authentication credentials there but I could test with an incorrect bucket name which yields this:

```
Exception in thread "main" com.google.cloud.storage.StorageException: The specified bucket does not exist.
	at com.google.cloud.storage.StorageException.translate(StorageException.java:165)
	at com.google.cloud.storage.spi.v1.HttpStorageRpc.translate(HttpStorageRpc.java:297)
	at com.google.cloud.storage.spi.v1.HttpStorageRpc.list(HttpStorageRpc.java:442)
	at com.google.cloud.storage.StorageImpl.lambda$listBlobs$7(StorageImpl.java:405)
	at com.google.api.gax.retrying.DirectRetryingExecutor.submit(DirectRetryingExecutor.java:103)
	at com.google.cloud.RetryHelper.run(RetryHelper.java:76)
	at com.google.cloud.RetryHelper.runWithRetries(RetryHelper.java:50)
	at com.google.cloud.storage.Retrying.run(Retrying.java:60)
	at com.google.cloud.storage.StorageImpl.listBlobs(StorageImpl.java:402)
	at com.google.cloud.storage.StorageImpl.list(StorageImpl.java:368)
	at com.google.cloud.storage.contrib.nio.CloudStoragePath.seemsLikeADirectoryAndUsePseudoDirectories(CloudStoragePath.java:127)
	at com.google.cloud.storage.contrib.nio.CloudStorageFileSystemProvider.readAttributes(CloudStorageFileSystemProvider.java:822)
	at java.base/java.nio.file.Files.readAttributes(Files.java:1765)
	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystemProvider.ensureFile(ZipFileSystemProvider.java:83)
	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystemProvider.newFileSystem(ZipFileSystemProvider.java:99)
	at java.base/java.nio.file.FileSystems.newFileSystem(FileSystems.java:337)
	at java.base/java.nio.file.FileSystems.newFileSystem(FileSystems.java:286)
	at com.google.edwmigration.dumper.application.dumper.MetadataDumper.run(MetadataDumper.java:262)
	at com.google.edwmigration.dumper.application.dumper.MetadataDumper.run(MetadataDumper.java:124)
	at com.google.edwmigration.dumper.application.dumper.MetadataDumper.main(MetadataDumper.java:408)
Caused by: com.google.api.client.googleapis.json.GoogleJsonResponseException: 404 Not Found
GET https://storage.googleapis.com/storage/v1/b/no-such-bucket123123/o?maxResults=1&prefix=foo/dwh-migration-redshift-metadata.zip/&projection=full
{
  "code" : 404,
  "errors" : [ {
    "domain" : "global",
    "message" : "The specified bucket does not exist.",
    "reason" : "notFound"
  } ],
  "message" : "The specified bucket does not exist."
}
```

A similar message should appear when the credentials are missing or insufficient.